Sources/NIOPosix/SocketChannel.swift

```diff
@@ -651,7 +651,7 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
                     metadata = nil
                 }
 
-                let msg = AddressedEnvelope(remoteAddress: rawAddress.convert(),
+                let msg = AddressedEnvelope(remoteAddress: try rawAddress.convert(),
```

Comment:
Can we lift the call to `convert` higher in the `.processed` block? Let's not trigger any side-effects until we know we're going to process the read.

Reply:
f7ca7428d